### ISSUE TITLE
No pnpm executable after recommended `npx pnpm add -g pnpm`

### ISSUE THREAD

**Issue by segrey** [pnpm/pnpm#2873]
### pnpm version:

```
$ npx pnpm --version
npx: installed 1 in 1.335s
5.6.1
```
### Code to reproduce the issue:

Have Node.js installed via nvm:
```
$ which node
/home/segrey/.nvm/versions/node/v14.8.0/bin/node
$ nvm current
v14.8.0
$ npm config get prefix
/home/segrey/.nvm/versions/node/v14.8.0
```

Install pnpm using a recommended way from https://pnpm.js.org/en/installation:
```
npx pnpm add -g pnpm
pnpm add -g pnpm
```
Output:
```
$ npx pnpm add -g pnpm
npx: installed 1 in 1.479s
Nothing to stop. No server is running for the store at /home/segrey/.pnpm-store/v3
Already up-to-date
Progress: resolved 1, reused 0, downloaded 0, added 0, done
$ pnpm add -g pnpm

Command 'pnpm' not found, did you mean:

  command 'npm' from deb npm

Try: sudo apt install <deb name>

$ npx pnpm config get prefix
npx: installed 1 in 1.447s
/home/segrey/.nvm/versions/node/v14.8.0
```

### Expected behavior:

`pnpm` executable is available in `PATH`

### Actual behavior:

`pnpm` executable is not available in `PATH`

### Additional information:

 - `node -v` prints: v14.8.0
 - Windows, OS X, or Linux?: Linux

**Comment by ZeroAurora** [comment]
I found that in `npx`, `pnpm` has a incorrect bin path:

```
zeroaurora@zeroaurora-fedora ~> npx pnpm bin -g
/home/zeroaurora/.npm/_npx/18421/bin
```

**Comment by zkochan** [comment]
@ZeroAurora good catch! pnpm should probably ignore the directories that contain `_npx`, when searching for a suitable global location.

**Comment by finico** [comment]
I still have this issue

```
> npx pnpm --version
npx: installed 1 in 1.105s
6.0.0
```

```
> npx pnpm add -g pnpm
npx: installed 1 in 1.037s
Nothing to stop. No server is running for the store at /Users/me/.pnpm-store/v3
Already up-to-date
Progress: resolved 1, reused 0, downloaded 0, added 0, done
```

```
> pnpm --version
zsh: command not found: pnpm
```

```
> node -v  
v14.16.0
> npm -v   
6.14.11
> sw_vers 
ProductName:	macOS
ProductVersion:	11.3
BuildVersion:	20E5224a
```

**Comment by zkochan** [comment]
What about installation with the standalone script? Does it work?

```
curl -f https://get.pnpm.io/v6.js | node - add --global pnpm
```

Alexey Gonchar ***@***.***> ezt írta (időpont: 2021. ápr. 6.,
K 22:04):

**Comment by zkochan** [comment]
@finico what is your npm version?

**Comment by finico** [comment]
@zkochan yes, it works. But installing via remote script through curl is not my favourite way to install

**Comment by zkochan** [comment]
but it should be done only once, for subsequent updates, you will just run `pnpm update -g pnpm`

**Comment by finico** [comment]
> but it should be done only once, for subsequent updates, you will just run `pnpm update -g pnpm`

it's not quite so if it's Docker

**Comment by yikyo** [comment]
@zkochan can you reopen this issue

I still have this issue

**Comment by jor3l** [comment]
Same issue on mac, even with the manual install I still get pnpm not found

**Comment by satphil** [comment]
I have this issue too in your otherwise most excellent code. 

Just doing `npx pnpm add -g pnpm` as user root omits these 3 files (at least):

```
/usr/bin/pnpm
/usr/pnpm-global/5/pnpm-lock.yaml
/root/.cache/pnpm/metadata/registry.npmjs.org/pnpm.json
```

**Comment by dfee** [comment]
I think I just ran into the same (or sibling) problem. I've just switched to [nix](https://nixos.org/download.html) on MacOS.

With Nix, I needed to specify the following `~/.npmrc`:
```
prefix = ${HOME}/.npm-packages
```

Now, I'm not sure *what* the problem is, but when attempting a `pnpm create next-app` I was receiving `ENOENT not found: pnpm` (whereas, `pnpm` itself was fine for some other commands, like `pnpm i`).

After `npm uninstall -g pnpm` and installing with the script, my problems seem to be resolved.
